torch/csrc/jit/tensorexpr/loopnest.h

```diff
@@ -418,9 +418,16 @@ class TORCH_API LoopNest {
   // Returns true if the given loop has a loop-carried dependence.
   static bool hasLoopCarriedDependence(ForPtr loop);
 
+  // Unrolls all the iterations of the given loop.
+  // Requires that the loop bounds are constant.
   static void unroll(ForPtr f, StmtPtr* unrolled);
   static void unroll(ForPtr f);
 
+  // Unrolls the given loop for the specified factor.
+  // This does not require constant bounds for the loop being unrolled.
+  static void unroll(ForPtr f, int factor, ForPtr* tail);
+  static void unroll(ForPtr f, int factor);
```

Comment:
Should we have a version of this method that also returns `StmtPtr* unrolled`? Also, it might be worth using different names for API for full (the original behavior) and partial (the added behavior) unrolling

Reply:
I'm not sure if returning `unrolled` is really needed here, since the given `For` is still present after unrolling. This is unlike the other `unroll` scenario where the input `For` will be gone after unrolling.

Reply:
I'm open to anything w.r.t. naming. How about `unrollAll` for the existing APIs and just `unroll` for this one?

Do you have any other suggestions?

Reply:
> I'm not sure if returning unrolled is really needed here, since the given For is still present after unrolling. 

What happens if we unroll a loop with a constant trip count 8 and request a factor of 8? If you don't think returning the unrolled thing is useful, I'm ok with the current version too.

> I'm open to anything w.r.t. naming. How about `unrollAll` for the existing APIs and just `unroll` for this one?

`unrollAll` can be read as "unroll all loops" rather than "unroll all iterations of a loop", which can be a bit misleading. How about we use `fullUnroll` or `completeUnroll` for the old one and `unroll` for the new one? We can also use `legacyUnroll` 😀

Reply:
> What happens if we unroll a loop with a constant trip count 8 and request a factor of 8? 

This will end up as a loop with a trip count of 1 with all the unrolled code inside it (because this API does not `simplify`). So, the `For` that was input will still be valid after `unroll`. If we see a need to `simplify` inside this API in future (not sure why that would happen), then we might have to return the `unrolled` loop. Leaving this as is for now.